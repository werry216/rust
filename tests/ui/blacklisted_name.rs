// Copyright 2014-2018 The Rust Project Developers. See the COPYRIGHT
// file at the top-level directory of this distribution.
//
// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
// option. This file may not be copied, modified, or distributed
// except according to those terms.





#![allow(dead_code, clippy::similar_names, clippy::single_match, clippy::toplevel_ref_arg, unused_mut, unused_variables)]
#![warn(clippy::blacklisted_name)]

fn test(foo: ()) {}

fn main() {
    let foo = 42;
    let bar = 42;
    let baz = 42;

    let barb = 42;
    let barbaric = 42;

    match (42, Some(1337), Some(0)) {
        (foo, Some(bar), baz @ Some(_)) => (),
        _ => (),
    }
}

fn issue_1647(mut foo: u8) {
    let mut bar = 0;
    if let Some(mut baz) = Some(42) {}
}

fn issue_1647_ref() {
    let ref bar = 0;
    if let Some(ref baz) = Some(42) {}
}

fn issue_1647_ref_mut() {
    let ref mut bar = 0;
    if let Some(ref mut baz) = Some(42) {}
}
