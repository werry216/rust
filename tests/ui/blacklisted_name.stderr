error: use of a blacklisted/placeholder name `foo`
  --> $DIR/blacklisted_name.rs:17:9
   |
17 | fn test(foo: ()) {}
   |         ^^^
   |
   = note: `-D clippy::blacklisted-name` implied by `-D warnings`

error: use of a blacklisted/placeholder name `foo`
  --> $DIR/blacklisted_name.rs:20:9
   |
20 |     let foo = 42;
   |         ^^^

error: use of a blacklisted/placeholder name `bar`
  --> $DIR/blacklisted_name.rs:21:9
   |
21 |     let bar = 42;
   |         ^^^

error: use of a blacklisted/placeholder name `baz`
  --> $DIR/blacklisted_name.rs:22:9
   |
22 |     let baz = 42;
   |         ^^^

error: use of a blacklisted/placeholder name `foo`
  --> $DIR/blacklisted_name.rs:28:10
   |
28 |         (foo, Some(bar), baz @ Some(_)) => (),
   |          ^^^

error: use of a blacklisted/placeholder name `bar`
  --> $DIR/blacklisted_name.rs:28:20
   |
28 |         (foo, Some(bar), baz @ Some(_)) => (),
   |                    ^^^

error: use of a blacklisted/placeholder name `baz`
  --> $DIR/blacklisted_name.rs:28:26
   |
28 |         (foo, Some(bar), baz @ Some(_)) => (),
   |                          ^^^

error: use of a blacklisted/placeholder name `foo`
  --> $DIR/blacklisted_name.rs:33:19
   |
33 | fn issue_1647(mut foo: u8) {
   |                   ^^^

error: use of a blacklisted/placeholder name `bar`
  --> $DIR/blacklisted_name.rs:34:13
   |
34 |     let mut bar = 0;
   |             ^^^

error: use of a blacklisted/placeholder name `baz`
  --> $DIR/blacklisted_name.rs:35:21
   |
35 |     if let Some(mut baz) = Some(42) {}
   |                     ^^^

error: use of a blacklisted/placeholder name `bar`
  --> $DIR/blacklisted_name.rs:39:13
   |
39 |     let ref bar = 0;
   |             ^^^

error: use of a blacklisted/placeholder name `baz`
  --> $DIR/blacklisted_name.rs:40:21
   |
40 |     if let Some(ref baz) = Some(42) {}
   |                     ^^^

error: use of a blacklisted/placeholder name `bar`
  --> $DIR/blacklisted_name.rs:44:17
   |
44 |     let ref mut bar = 0;
   |                 ^^^

error: use of a blacklisted/placeholder name `baz`
  --> $DIR/blacklisted_name.rs:45:25
   |
45 |     if let Some(ref mut baz) = Some(42) {}
   |                         ^^^

error: aborting due to 14 previous errors

